src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,198 @@
+
+                                        if (methodReferenceOperation == null)
+                                        {
+                                            return;
+                                        }
+
+                                        var methodSymbol = methodReferenceOperation.Method;
+                                        
+                                        if (!IsCertificateValidationFunction(methodSymbol, obj, x509Certificate, x509Chain, sslPolicyErrors))
+                                        {
+                                            return;
+                                        }
+
+                                        var blockOperation = methodSymbol.GetTopmostOperationBlock(compilation);
+
+                                        if (blockOperation == null)
+                                        {
+                                            return;
+                                        }
+
+                                        // TODO(LINCHE): This is an issue tracked by #2009. We filter extraneous based on IsImplicit.
+                                        var targetOperations = GetFilteredOperations(ImmutableArray.ToImmutableArray(blockOperation.Descendants()));
+                                        alwaysReturnTrue = AlwaysReturnTrue(targetOperations);
+                                        break;
+                                }
+
+                                if (alwaysReturnTrue)
+                                {
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        delegateCreationOperation.CreateDiagnostic(
+                                            Rule));
+                                }
+                            }
+                        },
+                        OperationKind.DelegateCreation);
+                });
+        }
+
+        private static bool IsCertificateValidationFunction(IMethodSymbol methodSymbol, INamedTypeSymbol obj, INamedTypeSymbol x509Certificate, INamedTypeSymbol x509Chain, INamedTypeSymbol sslPolicyErrors)
+        {
+            if (methodSymbol.ReturnType.SpecialType != SpecialType.System_Boolean)
+            {
+                return false;
+            }
+
+            var parameters = methodSymbol.Parameters;
+
+            if (parameters.Length != 4)
+            {
+                return false;
+            }
+            
+            if (!parameters[0].Type.Equals(obj)
+                || !parameters[1].Type.Equals(x509Certificate)
+                || !parameters[2].Type.Equals(x509Chain)
+                || !parameters[3].Type.Equals(sslPolicyErrors))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static IEnumerable<IOperation> GetFilteredOperations(ImmutableArray<IOperation> blockOperations)
+            => blockOperations.GetOperations().Where(o => !o.IsImplicit);
```

Comment:
As you cited in one of your previous comments, this will not handle the case where you have an expression bodied member returning true `(...) => true;`?

Reply:
Yes, but it happens when the delegation kind is "AnonymousFunction".

Reply:
Even regular methods can have expression bodies, not just lambdas (anonymous functions). Probably need an additional test where you have a method `bool M(...) => true;`